game/training/train.py

```diff
@@ -46,19 +48,38 @@ def evaluate_agent(
 
 
 def train_against_self(
-    discount: float,
-    n_episodes: int,
     agent: DeepVAgent,
+    path_to_save: str,
+    n_episodes: int = 1000,
+    discount: float = 1,
     n_test_runs: int = 10,
     num_opponents: int = 5,
     interval_test: int = 100,
+    checkpoint_interval: int = 500,
 ) -> Tuple[List[float], List[float]]:
+
     win_rates, losses = [[]] + [[0] for _ in range(num_opponents - 1)], []
     env = ConnectFourGymEnv()
     opponents = [RandomAgent(constants.PLAYER1, env.board.shape)]
 
+    os.makedirs(path_to_save, exist_ok=True)
+
+    progress_path = os.path.join(path_to_save, "progress.png")
+    config_path = os.path.join(path_to_save, "config.json")
+    agent_path = os.path.join(path_to_save, "agent.pt")
+
     period_change_opponent = (n_episodes // num_opponents) + 1
 
+    architecture = list(map(str, next(agent.value_network.children())))
+
+    config = {
+        "n_episodes": 0,
+        "discount": discount,
+        "epsilon": agent.epsilon,
+        "value_network": architecture,
+        "learning_rate": agent.learning_rate,
+    }
```

Comment:
We should also add this to the train not against self function, but we should be smart and find a way to somehow factorize it